Game selection feature

madagas-karrh opened the pull request:
# Summary of changes
Added a method in games.html that listens for certain keys being pushed.
Selected game is highlighted blue (its a little ugly)
Right now, it is not completely dynamic because it is based off my example of 5 games per row and only two rows. I am using this as a delimiter, so that the selection can't go out of bounds.

Added in extra methods for issue 31, user can now sort the games by name or by genre.
Pressing 'h' allows you to sort by name, pressing 'k' allows you to sort by genre
The code is kinda sloppy right now, but I will clean it up 


# How was this tested?
Tested on different browsers and sizes. The selection can not go out of bounds.

# TODO
Implement POST request
Make selection dynamic

#21 
#31

madagas-karrh commented:
okay I uncommented the actual pull request - it sends the path to the executable of the game selected. Is that what you want?

madagas-karrh commented:
i replaced the ajax request with axios,

starfys commented on arclightning_frontend/games.html:
```diff
@@ -27,17 +40,24 @@
+	    },
             data: {
+		selectedIndex: 0,
                 items: [
```
We'll need to restructure this as a dictionary

Something like
```js
{
  'smash':  {name: "Super Smash Bros", description: "crossover fighting game", genre: ["fighter", "crossover"], thumbnail_path: 'images/peach.png'},
  'touhou':  {name: "Touhou", description: "bullet hell with waifus", genre: ["bullet hell", "anime"], thumbnail_path: 'images/touhouWaifu.png'},
}
```
where a unique key id identifies each game and the value represents game info. Also remove exe_path, since the client doesn't need that info.

starfys commented on arclightning_frontend/games.html:
```diff
@@ -53,6 +73,62 @@ <h1>Arc Lightning</h1>
+	methods: {
+		sendSelectedGame: function() {
+			//check to see if game exists
+			var gameSelected = this.items[this.selectedIndex].exe_path
```
i wouldn't use exe_path. Once you refactor the data to look like the above comment, just post using the unique key id

starfys commented on arclightning_frontend/games.html:
```diff
@@ -53,6 +73,62 @@ <h1>Arc Lightning</h1>
+				}) */
+		},
+
+		onKeyDown(e){
```
is this the same function syntax as `document.addEventListener`? Let's rename `e` to `event` to be more clear

starfys commented on arclightning_frontend/games.html:
```diff
@@ -53,6 +73,62 @@ <h1>Arc Lightning</h1>
+		},
+
+		onKeyDown(e){
+			switch(e.keyCode){
```
I think `e.key` will work here, which has much more readable values (strings representing keys) than keycodes lol

madagas-karrh commented:
I restructured the array of dictionaries into a dictionary of dictionaries, the POST request now sends the unique identifier for the specific game selected. Removed exe_path from the games dictionary and renamed variables as requested

starfys commented:
Looks good! Just fix the merge conflicts and we can merge.

madagas-karrh commented:
I think I fixed the merge conflict, I just added the keydown event into the switch statement and swapped it from being the key code number and the key name as requested

madagas-karrh commented:
Okay, so I added in the functionality for the users to sort the games, I didn't know that I wouldnt be able to open another pull request with the same fork oops! So, the user can now sort games by name using the 'h' key and by genre using the 'k' key. (Issue 31 said use l for that, but pressing l takes the user back to the demo screen)

EricSciullo commented on arclightning_frontend/games.html:
```diff
@@ -27,41 +40,165 @@
+	}
+            },
+		
+	computed:{
```
I think you can get rid of this
https://github.com/MTSUMakers/ArcLightning/blob/916044c76b374c2fb9d27bba4ecc9364725fd0ef/arclightning_frontend/games.html#L77-L93

madagas-karrh replied:
yes you are so efficient

EricSciullo commented on arclightning_frontend/games.html:
```diff
@@ -27,41 +40,165 @@
+				resultDict[sortedArr[i]] = this.items[sortedArr[i]];
+			}
+			this.items = resultDict;
+			return resultDict;			
```
I don't think you have to return anything here since you are just trying to change the `items` dictionary which is done in the previous line. Same for the sortGenre() function.

madagas-karrh replied:
oh you are CORRECT !!! thank you

madagas-karrh commented:
I think this should be good to go now, let me know if there are any other issues to be fixed
